### PR TITLE
Add command line args for enableVisualizer() on SimulatedTests

### LINKED ISSUE #1751: Add command line args for enableVisualizer() on SimulatedTests
### Description of the task
In order to visualize the simulated tests, we need to manually add `enableVisualizer()` to each test. It would be great to be able to enable this via command line, e.g. `bazel test //software/simulated_tests:corner_kick_play_test --enable_visualizer`. 

We can do this by removing the `gtest_main` dependency from the BUILD file and adding a custom main that sets a flag (dynamic param?) depending on the command line args. https://stackoverflow.com/questions/4818785/how-to-pass-parameters-to-the-gtest

This main should be refactored into a header, perhaps `simulated_test_fixture.h`, so that all `SimulatedTests` can share it
  

### Acceptance criteria

- [x] main function that takes command line args and sets a flag
- [x] change all simulated tests to enable visualization via command line arg by respecting the flag

### PR DESCRIPTION
**Pull request by itsarune**
### Description

- Added a custom gtest main that is run whenever simulated tests run
- updated the build files and replaced references to standard main to the new one
- added a yaml so that we can pass in `--enable_visualizer` on the command line

There are two ways to run the tests in the visualizer:
1. `bazel test //software/simulated_tests:some-target --test-arg="--enable_visualizer"`
2. `bazel run //software/simulated_tests:some-target -- --enable_visualizer`

Running without the flag runs the test normally.

### Testing Done

- current tests pass
- verified that the `--enable_visualizer` flag works for all tests
- there is an error with `free_kick_play` when running `--enable_visualizer` on it but it is unrelated for the scope of this ticket

### Resolved Issues

resolves #1751

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — COMMENTED** [review]
Tested locally, works really well for me, nice work! :100:

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Good work. It would be nice to put the commands in `docs/getting-started.md` as well

**Review by jared-paul — COMMENTED** [review]
LGTM aside from Jon's comment, I agree with adding it to SimulatedTestFixture. Will reapprove once that's done

File: src/software/simulated_tests/corner_kick_play_test.cpp [hunk 1]
```diff
@@ -19,6 +19,10 @@ class CornerKickPlayTest : public SimulatedTestFixture
 
 TEST_F(CornerKickPlayTest, test_corner_kick_play)
 {
+    if (SimulatedTestFixture::enable_visualizer)
+    {
+        enableVisualizer();
+    }
```
[the remark this answers is not in the document]
Comment (itsarune) [reply src/software/simulated_tests/corner_kick_play_test.cpp]: added!

**Review by jonathanlew — DISMISSED** [review]
LGTM pending fix formatting

**Review by jonathanlew — DISMISSED** [review]

**Review by akhilveeraghanta — DISMISSED** [review]
💯

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**